virt-v2v/cold/entrypoint.go

```diff
@@ -0,0 +1,274 @@
+	case OVA:
+		virtV2vArgs = append(virtV2vArgs, "-i", "ova", os.Getenv("V2V_diskPath"))
+	}
+
+	fmt.Println("Preparing virt-v2v")
+	if err := os.MkdirAll(DIR, os.ModePerm); err != nil {
+		fmt.Println("Error creating directory  ", err)
+		os.Exit(1)
+	}
+	virtV2vArgs = append(virtV2vArgs, "-o", "local", "-os", DIR)
+
+	//Disks on filesystem storage.
+	if err := LinkDisks(FS, 15); err != nil {
+		os.Exit(1)
+	}
+	//Disks on block storage.
+	if err := LinkDisks(Block, 10); err != nil {
+		os.Exit(1)
+	}
+
+	if err := executeVirtV2v(virtV2vArgs); err != nil {
+		os.Exit(1)
+	}
+
+	if source == OVA {
+		var err error
+		xmlFilePath, err = waitForXMLFile(DIR, "xml", 5, 60)
+		if err != nil {
+			fmt.Println("Error waiting for XML file:", err)
+			os.Exit(1)
+		}
+
+		http.HandleFunc("/firmware", firmwareHandler)
+		server := &http.Server{Addr: ":8080"}
+
+		go func() {
+			// Wait for the first request to be processed
+			<-requestProcessed
+			fmt.Println("Shutting down server.")
+			if err := server.Shutdown(context.Background()); err != nil {
+				fmt.Printf("Error shutting down server: %v\n", err)
+				os.Exit(1)
+			}
+		}()
+
+		fmt.Println("Starting server on :8080")
+		if err := server.ListenAndServe(); err != http.ErrServerClosed {
+			fmt.Printf("Error starting server: %v\n", err)
+			os.Exit(1)
+		}
+
+		fmt.Println("Server stopped")
+	}
+}
+
+func checkEnvVariablesSet(envVars ...string) bool {
+	for _, v := range envVars {
+		if os.Getenv(v) == "" {
+			return false
+		}
+	}
+	return true
+}
+
+func genName(diskNum int) string {
+
+	if diskNum <= 0 {
+		return ""
+	}
+
+	letters := []int32("abcdefghijklmnopqrstuvwxyz")
+	index := (diskNum - 1) % len(letters)
+	cycels := (diskNum - 1) % len(letters)
+
+	return genName(cycels) + string(letters[index])
+}
+
+func LinkDisks(diskKind string, num int) (err error) {
+	disks, err := filepath.Glob(diskKind)
+	if err != nil {
+		fmt.Println("Error getting disks ", err)
+		return
+	}
+
+	for _, disk := range disks {
+		diskNum, err := strconv.Atoi(disk[num:])
+		if err != nil {
+			fmt.Println("Error geting disks names ", err)
+			return err
+		}
+		diskLink := fmt.Sprintf("%s/%s-sd%s", DIR, os.Getenv("V2V_vmName"), genName(diskNum+1))
+		diskImgPath := disk
+		if diskKind == FS {
+			diskImgPath = fmt.Sprintf("%s/disk.img", disk)
+		}
+		if err = os.Symlink(diskImgPath, diskLink); err != nil {
+			fmt.Println("Error creating disk link ", err)
+			return err
+		}
+	}
+	return
+}
+
+func executeVirtV2v(args []string) (err error) {
+	virtV2vCmd := exec.Command("bash", "-c", strings.Join(args, " "), "|& /usr/local/bin/virt-v2v-monitor")
+	virtV2vCmd.Stdout = os.Stdout
+	virtV2vCmd.Stderr = os.Stderr
+	if err = virtV2vCmd.Run(); err != nil {
+		fmt.Printf("Error executing command: %v\n", err)
+		return
+	}
+	return
+}
+
+func waitForXMLFile(dir, fileExtension string, timeoutMinutes, checkIntervalSeconds int) (string, error) {
+	deadline := time.Now().Add(time.Duration(timeoutMinutes) * time.Minute)
```

Comment:
on which step (area) is it generated by virt-v2v? are we sure 5 minutes is enough/too much?

Reply:
I drooped this entirely, there another comment about that below.